### ISSUE TITLE
Not able to configure on hassio 0.108.5 with HACS

### ISSUE THREAD

**Issue by souravj96** [tprelog/homeassistant-gmusic_player#10]
I'm using HACS for installation
after installation
created `packages` directory inside `config`
and paste this code in `configuration.yaml`
```
homeassistant:
  packages: !include_dir_named packages
```
and created these two files inside the `packages` directory
[gmusic_config.yaml](https://raw.githubusercontent.com/tprelog/homeassistant-gmusic_player/master/packages/gmusic_config.yaml)
[gmusic_player.yaml](https://raw.githubusercontent.com/tprelog/homeassistant-gmusic_player/master/packages/gmusic_player.yaml)
and after restart i got this error log on hassio log
```
2020-04-18 02:41:39 ERROR (SyncWorker_10) [custom_components.gmusic_player.media_player] Failed legacy log in, check http://unofficial-google-music-api.readthedocs.io/en/latest/reference/mobileclient.html#gmusicapi.clients.Mobileclient.login
2020-04-18 02:41:39 ERROR (MainThread) [homeassistant.components.media_player] Error while setting up gmusic_player platform for media_player
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/gmusic_player/media_player.py", line 121, in setup_platform
    add_devices([GmusicComponent(hass, config)])
TypeError: __init__() should return None, not 'bool'
2020-04-18 02:41:48 WARNING (MainThread) [homeassistant.components.template.sensor] Could not render template gmusic_receiver, the state is unknown.
```
please help me out

**Comment by tprelog** [comment]
```
2020-04-18 02:41:39 ERROR (SyncWorker_10) [custom_components.gmusic_player.media_player] Failed legacy log in, check http://unofficial-google-music-api.readthedocs.io/en/latest/reference/mobileclient.html#gmusicapi.clients.Mobileclient.login
```

Looks like your not logging in. Please set and/or double check your username and password in `gmusic_config.yaml` or your `secrets.yaml`

**Comment by souravj96** [comment]
it's google account email id as username and google's password as password, right?
and my account is two-step verification enabled, is it any problem??

**Comment by tprelog** [comment]
Yep. If your using 2FA, you'll need to setup an app password in your Google account.

This link can help you with that:
https://support.google.com/accounts/answer/185833?hl=en

**Comment by souravj96** [comment]
it helps a lot. 
now I'm logged in. but getting the device_id error. what would be the device id?
and how to attach the device id with my google account?

here's the log
```
2020-04-18 11:18:23 ERROR (MainThread) [homeassistant.components.media_player] Error while setting up gmusic_player platform for media_player
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/gmusic_player/media_player.py", line 121, in setup_platform
    add_devices([GmusicComponent(hass, config)])
  File "/config/custom_components/gmusic_player/media_player.py", line 172, in __init__
    logged_in = self._api.login(_username, _password, _device_id, authtoken)
  File "<decorator-gen-76>", line 2, in login
  File "/usr/local/lib/python3.7/site-packages/gmusicapi/utils/utils.py", line 128, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gmusicapi/clients/mobileclient.py", line 215, in login
    return self._login(session_login, android_id, locale)
  File "/usr/local/lib/python3.7/site-packages/gmusicapi/clients/mobileclient.py", line 138, in _login
    self.android_id = self._validate_device_id(device_id, is_mac=is_mac)
  File "/usr/local/lib/python3.7/site-packages/gmusicapi/clients/mobileclient.py", line 75, in _validate_device_id
    raise InvalidDeviceId('Invalid device_id %s.' % device_id, device_ids)
gmusicapi.exceptions.InvalidDeviceId: Invalid device_id Android.It looks like your account does not have any 
2020-04-18 11:18:31 WARNING (MainThread) [homeassistant.components.template.sensor] Could not render template gmusic_receiver, the state is unknown.
```

**Comment by tprelog** [comment]
Please see the main thread for this in the Home Assistant Forums.

https://community.home-assistant.io/t/google-music-in-ha-now-as-a-media-player/119918

![image](https://user-images.githubusercontent.com/35702532/79640353-2e7c5d80-815f-11ea-8374-24b5818cc944.png)

**Comment by souravj96** [comment]
I put 00 but as device_id but not getting the list of device ids.
```
  File "/usr/local/lib/python3.7/site-packages/gmusicapi/clients/mobileclient.py", line 75, in _validate_device_id
    raise InvalidDeviceId('Invalid device_id %s.' % device_id, device_ids)
gmusicapi.exceptions.InvalidDeviceId: Invalid device_id 0.It looks like your account does not have any 
```
is there any way to generate device id?

**Comment by tprelog** [comment]
Do you have any sort of android device ( phone or tablet)? If you install the Google Music App, it will register a device ID

https://play.google.com/store/apps/details?id=com.google.android.music

**Comment by souravj96** [comment]
yes, I have an android mobile and there showing no device authorized in settings -> manage devices section. 
but my mobile is logged in and registered with google play music.
really getting confused.....

**Comment by souravj96** [comment]
it's working now.
thank you for your support @tprelog

**Comment by souravj96** [comment]
@achille858 yes i solved my issue

**Comment by achille858** [comment]
> @achille858 yes i solved my issue

how did you solve? What was the problem??

**Comment by souravj96** [comment]
many problems were there.
1. my google account was 2 step verification enabled. so I had to create an app password.
 follow this link [https://community.home-assistant.io/t/google-music-in-ha-now-as-a-media-player/119918](https://community.home-assistant.io/t/google-music-in-ha-now-as-a-media-player/119918)
2. device id was unavailable. first, I put 00 as device_id but it returns nothing. then I go to google music manage device section, there was no device authorized. then I added my android mobile as an authorized device. now I get device_id when I put 00 and copy the id.
finally, it was the moment I got no errors on the HA log.
